xds/internal/balancer/outlierdetection/config.go

```diff
@@ -154,8 +153,10 @@ type LBConfig struct {
 	ChildPolicy *internalserviceconfig.BalancerConfig `json:"childPolicy,omitempty"`
 }
 
-// Equal returns whether the LBConfig is the same with the parameter.
-func (lbc *LBConfig) Equal(lbc2 *LBConfig) bool {
+// EqualODOnly returns whether the LBConfig is same with the parameter outside
+// of the child policy, only comparing the Outlier Detection specific
+// configuration.
+func (lbc *LBConfig) EqualODOnly(lbc2 *LBConfig) bool {
```

Comment:
Also, we should probably have a test of this function that will fail if we add new fields to the config struct.

Even a purely change-detector test would help; something like:

```go
func TestEqualFields(t *testing.T) {
	fields := map[string]bool{"LoadBalancingConfig": true, "Interval": true, "BaseEjectionTime": true, ....}
	t := reflect.TypeOf((*LBConfig)(nil))
	for i := 0; i < t.NumField(); i++ { // that's probably not the right method name
		if n := t.Field(i).Name; !fields[n] {
			t.Error("New field %q; update this test and EqualIgnoringChildPolicy", n)
		}
	}
}
```

Reply:
Added this change-detector test for every node in the config tree.